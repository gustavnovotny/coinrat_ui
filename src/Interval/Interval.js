// @flow
class Interval {
  since = null
  till = null

  constructor(since: ?Date = null, till: ?Date = null) {
    if (since !== null && till !== null && since > till) {
      throw Error('Since must be < than till')
    }

    this.since = since
    this.till = till
  }

  toIso = () => {
    return {
      since: this.since !== null ? this.since.toISOString() : null,
      till: this.till !== null ? this.till.toISOString() : null,
    }
  }

  withClosedFromRight = (date: Date) => {
    if (this.till === null) {
      const till = date > this.since ? date : this.since
      return new Interval(this.since, till)
    }
    return new Interval(this.since, this.till)
  }

  isEmpty = () => {
    return this.since.getTime() === this.till.getTime()
  }
}


export default Interval
