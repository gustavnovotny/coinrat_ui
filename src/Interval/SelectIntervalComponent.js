// @flow
import React, {Component} from "react"
import {DateRangeInput} from "@blueprintjs/datetime"
import {Label} from "@blueprintjs/core"
import Interval from "./Interval"
import appMainToaster from "../Toaster"
import {Box, Flex} from "reflexbox"

type Props = {
  defaultSelectedInterval: Interval,
  onChange: (interval: Interval) => void,
}

const lineStyles = {
  verticalAlign: 'middle',
  lineHeight: 30 + 'px',
  display: 'inline',
  marginLeft: 7 + 'px',
  minWidth: 120 + 'px',
}

class SelectIntervalComponent extends Component<Props> {
  props: Props

  handleChange = (data: Array<?Date>) => {
    let since = data[0]
    let till = data[1]
    since = Object.prototype.toString.call(since) === '[object Date]' && !isNaN(since.getTime()) ? since : null
    till = Object.prototype.toString.call(till) === '[object Date]' && !isNaN(till.getTime()) ? till : null

    if (since !== null && till !== null && since > till) {
      appMainToaster.show({message: "Since must be < than till.", className: 'pt-intent-danger'})
      return
    }

    this.props.onChange(new Interval(since, till))
  }

  render() {
    const interval = this.props.defaultSelectedInterval
    return <Flex>
      <Box w={120} style={{textAlign: 'right', ...lineStyles}}>
        <Label text='Interval:'/>
      </Box>
      <Box style={{textAlign: 'left', ...lineStyles}}>
        <DateRangeInput
          format="YYYY-MM-DD HH:mm:ss"
          value={[interval.since, interval.till]}
          onChange={this.handleChange}
          allowSingleDayRange={true}
        />
      </Box>
    </Flex>
  }
}

export default SelectIntervalComponent
