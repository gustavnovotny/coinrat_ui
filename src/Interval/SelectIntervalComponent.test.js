// @flow
import React from 'react'
import renderer from 'react-test-renderer'
import SelectIntervalComponent from "./SelectIntervalComponent"
import Interval from "./Interval"

it('renders correctly', () => {
  const tree = renderer
    .create(
      <SelectIntervalComponent
        defaultSelectedInterval={new Interval(new Date(2018, 1, 1, 1))}
        onChange={() => undefined}
        flashMessageHandler={() => undefined}
      />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
