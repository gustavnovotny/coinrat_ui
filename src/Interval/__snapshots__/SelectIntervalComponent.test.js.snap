// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="_Rfx0_"
>
  <div
    className="form-item form-label _Rfx1_"
  >
    <div
      className="pt-label"
    >
      Interval:
      <span
        className="pt-text-muted"
      />
    </div>
  </div>
  <div
    className="form-item form-element"
  >
    <span
      className="pt-popover-target"
      onClick={[Function]}
    >
      <div
        className="pt-control-group"
        tabIndex={undefined}
      >
        <div
          className="pt-input-group"
        >
          <input
            autoComplete="off"
            className="pt-input"
            disabled={false}
            onBlur={[Function]}
            onChange={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            placeholder="Start date"
            style={
              Object {
                "paddingRight": 0,
              }
            }
            type="text"
            value="2018-01-01 00:00:00"
          />
        </div>
        <div
          className="pt-input-group"
        >
          <input
            autoComplete="off"
            className="pt-input"
            disabled={false}
            onBlur={[Function]}
            onChange={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            placeholder="End date"
            style={
              Object {
                "paddingRight": 0,
              }
            }
            type="text"
            value=""
          />
        </div>
      </div>
    </span>
  </div>
  <div
    className="form-item form-suffix"
  >
    
     
    
  </div>
</div>
`;
